data/src/main/java/com/linkedin/data/template/WrappingArrayTemplate.java

```diff
@@ -162,7 +161,17 @@ protected E cacheLookup(Object object, int index) throws TemplateOutputCastExcep
     return wrapped;
   }
 
-  protected final Constructor<E> _constructor;
+  protected E coerceOutput(Object value) throws TemplateOutputCastException
+  {
+    if (_constructor == null)
+    {
+      _constructor = DataTemplateUtil.templateConstructor(_elementClass, schema().getItems());
+    }
+
+    return DataTemplateUtil.wrap(value, _constructor);
+  }
+
+  private Constructor<E> _constructor;
```

Comment:
I prefer keeping it private because, in case folks were using it via subclassing the change would break at build time. If we make it protected and folks continue to use it, there is a chance it may barf and throw NPE at runtime, since we now initialize the constructor on demand.